app/src/main/java/com/zulip/android/models/Stream.java

```diff
@@ -120,9 +120,8 @@ public static Stream getByName(ZulipApp app, String name) {
             if (stream == null) {
                 Log.w("Stream.getByName",
                         "We received a stream message for a stream we don't have data for. Fake it until you make it.");
-                stream = new Stream(name);
-                app.getDao(Stream.class).createIfNotExists(stream);
-            }
+                // catch the new stream object with auto-generated id
+                stream = app.getDao(Stream.class).createIfNotExists(new Stream(name));            }
```

Comment:
I know the stream objects have id given from server which are deserialized into id field in stream class here. I guess my concern is if the stream is null after querying for it in the database, we would be making a new stream object for it with id zero and store it in the database. Eventually, other stream objects generated this way would also have id 0 as well. Storing these in the database might cause a unique constraint exception. Am I mistake? I just want to clarify this doubt :)